### PR TITLE
Fix #57910 - Make autofetch period customizable

### LINKED ISSUE #57910: Git: Make autofetch period customizable
Hi

I would love to customize `git.autofetch`'s period in the settings instead of hardcoding three minutes:

https://github.com/Microsoft/vscode/blob/cd7c7c5fa2c16c6e8281436c35e9a7709cfbd89d/extensions/git/src/autofetch.ts#L22

Thanks!

### PR DESCRIPTION
**Pull request by skprabhanjan**
@joaomoreno , Reopening this PR which fixes #57910 , as it was not viewed upon and was kind of lost . 
Can you please review this and let me know if this is okay?
Thanks :)

### CODE REVIEW

**Comment by joaomoreno** [comment]
@skprabhanjan It was definitely not lost. Please be patient, we will get to the PRs in due time.

**Comment by skprabhanjan** [comment]
@joaomoreno , sorry for that , I will surely wait :)

File: extensions/git/src/autofetch.ts [hunk 1]
```diff
@@ -70,10 +73,15 @@ export class AutoFetcher {
 
 	private onConfiguration(): void {
 		const gitConfig = workspace.getConfiguration('git');
+		const Period = gitConfig.get<number>('autoFetchPeriod') as number;
```
Comment (kumarharsh) [inline extensions/git/src/autofetch.ts]:
```suggestion
		const period = gitConfig.get<number>('autoFetchPeriod');
```

Comment (kumarharsh) [inline extensions/git/src/autofetch.ts]:
```suggestion
		const period = gitConfig.get<number>('autoFetchPeriod');
```

File: extensions/git/src/autofetch.ts [hunk 2]
```diff
@@ -70,10 +73,15 @@ export class AutoFetcher {
...
 
 		if (gitConfig.get<boolean>('autofetch') === false) {
 			this.disable();
 		} else {
+			if (Period !== this.autoFetchPeriod) {
+				this.disable();
+				this.autoFetchPeriod = Period;
```
Comment (kumarharsh) [inline extensions/git/src/autofetch.ts]:
```suggestion
				this.autoFetchPeriod = period;
```

File: extensions/git/src/autofetch.ts [hunk 3]
```diff
@@ -70,10 +73,15 @@ export class AutoFetcher {
 
 	private onConfiguration(): void {
 		const gitConfig = workspace.getConfiguration('git');
+		const Period = gitConfig.get<number>('autoFetchPeriod') as number;
 
 		if (gitConfig.get<boolean>('autofetch') === false) {
 			this.disable();
 		} else {
+			if (Period !== this.autoFetchPeriod) {
```
Comment (kumarharsh) [inline extensions/git/src/autofetch.ts]:
```suggestion
			if (period !== this.autoFetchPeriod) {
```

**Comment by skprabhanjan** [comment]
@joaomoreno , Any updates on this ? 
The reviews were also verified

**Comment by skprabhanjan** [comment]
@joaomoreno, the related issue was closed , any reason or updates on this ?
Thanks :)

**Comment by joaomoreno** [comment]
Oh yeah absolutely. I ended up merging this one: https://github.com/Microsoft/vscode/pull/59988

Sorry for that @skprabhanjan! Thanks any way! :beers:

**Pull request closed without merging**